kotlin/display-utility-associations/src/main/java/com/esri/arcgisruntime/sample/displayutilityassociations/MainActivity.kt

```diff
@@ -0,0 +1,166 @@
+
+  // max scale at which to create graphics for the associations
+  private val maxScale = 2000
+
+  // create the utility network
+  private val utilityNetwork =
+    UtilityNetwork("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")
+
+  // overlay to hold graphics for all of the associations
+  private val associationsOverlay by lazy { GraphicsOverlay() }
+
+  // create a green dotted line symbol for attachment
+  private val attachmentSymbol by lazy {
+    SimpleLineSymbol(
+      SimpleLineSymbol.Style.DOT,
+      Color.GREEN,
+      5.0f
+    )
+  }
+
+  // create a red dotted line symbol for connectivity
+  private val connectivitySymbol by lazy {
+    SimpleLineSymbol(
+      SimpleLineSymbol.Style.DOT,
+      Color.RED,
+      5.0f
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    mapView.apply {
+      // add a topographic vector map with a viewpoint at several utility network associations
+      map = ArcGISMap(Basemap.Type.TOPOGRAPHIC_VECTOR, 41.8057655, -88.1489692, 23)
+
+      // add the a graphics overlay to hold association graphics
+      graphicsOverlays.add(associationsOverlay)
+    }
+
+    // load the utility network
+    utilityNetwork.loadAsync()
+    utilityNetwork.addDoneLoadingListener {
+
+      // get all of the edges and junctions in the network
+      val edges =
+        utilityNetwork.definition.networkSources.filter { it.sourceType == UtilityNetworkSource.Type.EDGE }
+      val junctions =
+        utilityNetwork.definition.networkSources.filter { it.sourceType == UtilityNetworkSource.Type.JUNCTION }
+
+      // add all edges that are not subnet lines to the map
+      edges.filter { it.sourceUsageType != UtilityNetworkSource.UsageType.SUBNET_LINE }
+        .forEach { source ->
+          mapView.map.operationalLayers.add(FeatureLayer(source.featureTable))
+        }
+      // add all junctions to the map
+      junctions.forEach { source ->
+        mapView.map.operationalLayers.add(FeatureLayer(source?.featureTable))
+      }
+
+      // populate the legend in the UI
+      val attachmentSwatchFuture = attachmentSymbol.createSwatchAsync(this, Color.TRANSPARENT)
+      attachmentSwatch.setImageBitmap(attachmentSwatchFuture.get())
+      val connectSwatchFuture = connectivitySymbol.createSwatchAsync(this, Color.TRANSPARENT)
+      connectivitySwatch.setImageBitmap(connectSwatchFuture.get())
+
+      // add association graphics at the initial view point
+      addAssociationGraphicsAsync()
+
+      // listen for navigation changes
+      mapView.addNavigationChangedListener {
+        // add association graphics for viewpoint after navigation change
+        addAssociationGraphicsAsync()
+      }
+    }
+  }
+
+  /**
+   * If greater than the maxScale, add association graphics for the map view's current extent.
```

Comment:
This seems like a conflicting message, since if >= maxScale you just return. Personally I would probably remove the start of this sentence, because I think the first comment where you handle the scale stuff describes it anyway.

```suggestion
   * Add association graphics for the map view's current extent.
```